### ISSUE TITLE
Need country flag feature enabled and some improvements

### ISSUE THREAD

**Issue by hassan31** [graycampbell/GCCountryPicker#21]
Hi,

This is a very good library and its easy to use and simple one. But what about country flags feature to be enabled in this?

And one more thing it would be nice if you give the customization of search controller, like in my case I've navigation bar color of black and search bar text color is also black, I saw you have declared it private and unable to change. However, I can change using manual integration.

Let me know if you want to enable this feature and I'll create a PR.

Regards,
Hassan

**Comment by graycampbell** [comment]
Thanks for the kind words and suggestions. I'm certainly open to adding country flags - I think it would be a good addition. I'm going to do a little research to make sure I can find the highest quality images for the flags.

Just to clarify your second point about search controller customization so I understand fully - would the issue you're encountering be fixed by simply making the `searchController` property on `GCCountryPickerViewController` public instead of private? Are you just trying to access the `searchBar` property of `searchController` so that you can change the appearance of the search bar?

**Comment by hassan31** [comment]
Hi,

I've my own customized theme for the navigation bar, please see attached `Screenshot_1` that is background color and white for its tint color. that looks great so far now.

<img width="475" alt="Screenshot_1" src="https://user-images.githubusercontent.com/3063382/57151755-b2e39e00-6d9f-11e9-8f66-50d85883563a.png">

Now see Screenshot_2, in which I've typed the country name 'Canada' but I can't see it as its text color is white. And also see the cancel button is in blue color and the tint color of the searchBar text is also in blue, As my theme is something dark wooden color with white, so that's why I need these changes. IIf you can customize your library more that would be great.

<img width="468" alt="Screenshot_2" src="https://user-images.githubusercontent.com/3063382/57151893-0e159080-6da0-11e9-9e1c-e3d1e2345022.png">

Look forward to hearing from you soon. Thanks

**Comment by graycampbell** [comment]
Would you also like to be able to customize the appearance of the table view? Allowing you to customize the appearance of the search bar would really just be as simple as making the `searchController` property public instead of private, but allowing you to customize the appearance of the table view would require more work.

**Comment by hassan31** [comment]
Yes, I think, there should be options to show the country flag before the country name and options to hide toggle country code, etc.

**Comment by hassan31** [comment]
And, also I'd like you to introduce the currency feature as well in this library. Because I don't want to use multiple pods for the same kind of purpose. In my project, I need currency, country code, country picker with the flag, etc.

**Comment by hassan31** [comment]
you can introduce enum like a country picker, currency picker, country code picker with its flag.

**Comment by graycampbell** [comment]
Just to clarify, when you say country code, do you mean calling code? As in the number that you put at the beginning of a phone number? Or do you mean the ISO 3166-2 country code that is unique to each country?

**Comment by hassan31** [comment]
> Just to clarify, when you say country code, do you mean calling code? As in the number that you put at the beginning of a phone number? Or do you mean the ISO 3166-2 country code that is unique to each country?

Yes Calling Code
